server/routerlicious/packages/services-client/src/test/restwrapper.spec.ts

```diff
@@ -82,6 +175,52 @@ describe("RestWrapper", () => {
             );
         });
 
+        it("429 Response Code should reject Promise with 0 retryAfter", async () => {
+            // arrange
+            const rw = new RestWrapper(baseurl, {}, {}, false, maxContentLength, axiosTooManyRequestsErrorZeroRetryAfterMock as AxiosInstance);
+
+            // act/assert
+            await rw.get(requestUrl).then(
+                // tslint:disable-next-line:no-void-expression
+                () => assert.fail("Promise was not rejected"),
+                // tslint:disable-next-line:no-void-expression
+                (err) => assert.ok(err, "Invalid response code rejected Promise"),
+            );
+        });
+
+        it("429 Response Code should reject Promise with negative retryAfter", async () => {
+            // arrange
+            const rw = new RestWrapper(baseurl, {}, {}, false, maxContentLength, axiosTooManyRequestsErrorNegativeRetryAfterMock as AxiosInstance);
+
+            // act/assert
+            await rw.get(requestUrl).then(
+                // tslint:disable-next-line:no-void-expression
+                () => assert.fail("Promise was not rejected"),
+                // tslint:disable-next-line:no-void-expression
+                (err) => assert.ok(err, "Invalid response code rejected Promise"),
+            );
+        });
+
+
+        it("429 Response Code should not reject Promise with positive retryAfter", async () => {
```

Comment:
This test should also assert that it is returning the result of the retry

Reply:
Thank you so much for offline discussion and pointing me to 'axios-mock-adapter'!